README.md

```diff
@@ -207,23 +220,67 @@ More information can be found [here](http://www.diveintopython3.net/strings.html
 | pyVersion        | Returns the version of Python                      | `pyVersion()`                                                        |
 
 
-# Usage Examples   
+# Wrapping python packages
+
+While using the original PythonInR to wrap a python package in R, automatically generates R functions and documentation, we added a few helper functions in out version of PythonEmbedInR:
+
+## Determine args and kwargs (determineArgsAndKwArgs)
+
+There are many ways to wrap a Python function in R. To create an R wrapper on top of the Python function without making any changes in the function signature, we need to figure out the user's input and pass them along to the Python function. This helper function is used to determine the parameters to pass to the underlying Python function.
+
+| example                                             | output                                                      |
+| `determineArgsAndKwArgs()`                          | `list(args=list(), kwargs=list())`                          |
+| `determineArgsAndKwArgs("foo", "bar")`              | `list(args=list("foo", "bar"), kwargs=list())`              |
+| `determineArgsAndKwArgs("foo", "bar", x="baz")`     | `list(args=list("foo", "bar"), kwargs=list(x="baz"))`       |
+| `determineArgsAndKwArgs(a="foo", b="bar", c="baz")` | `list(args=list(), kwargs=list(a="foo", b="bar", c="baz"))` |
+
+## Clean up stack trace (cleanUpStackTrace)
```

Comment:
cleanUpStackTrace() is really just a small utility which is part of the larger framework for generating wrapper functions. Do you think it makes more sense to add the larger framework here? cleanUpStackTrace() would then just be an internal method and there'd be no need for extensive documentation.